### PR TITLE
fix pair getfield pullback

### LINKED ISSUE #817: Cannot differentiate through dictionary
I am trying to replicate this example in the Zygote homepage (https://fluxml.ai/Zygote.jl/latest/#Structs-and-Types-1), but replacing the `struct` with `Dict`

```julia
p1 = Dict(:x=>1.0, :y=>2.0)
p2 = Dict(:x=>1.0, :y=>3.0)
function add_d(x,y)
	Dict(:x => (x[:x] + y[:x]), :y => (x[:y] + y[:y]))
end
dist(p::Dict) = p[:x]^2 + p[:y]^2
gradient(a -> dist(add_d(a, p1)), p2)[1])
```

But I am getting this error
```julia
MethodError: no method matching zero(::Symbol)

Closest candidates are:

zero(!Matched::Type{Pkg.Resolve.FieldValue}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Pkg/src/Resolve/fieldvalues.jl:38

zero(!Matched::Type{Dates.Date}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Dates/src/types.jl:405

zero(!Matched::Type{Dates.Time}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Dates/src/types.jl:406

...

    (::Zygote.var"#pair_getfield#294"{Pair{Symbol,Float64},Int64})(::Float64)@base.jl:136
    #2047#back@adjoint.jl:49[inlined]
    indexed_iterate@pair.jl:50[inlined]
    literal_indexed_iterate@builtins.jl:17[inlined]
    (::typeof(∂(literal_indexed_iterate)))(::Tuple{Float64,Nothing})@interface2.jl:0
    Dict@dict.jl:102[inlined]
    (::typeof(∂(Dict{Symbol,Float64})))(::Dict{Any,Any})@interface2.jl:0
    Dict@dict.jl:123[inlined]
    (::typeof(∂(Dict)))(::Dict{Any,Any})@interface2.jl:0
    add_d@Other: 2[inlined]
    (::typeof(∂(add_d)))(::Dict{Any,Any})@interface2.jl:0
    #1@Local: 1[inlined]
    (::typeof(∂(#1)))(::Float64)@interface2.jl:0
    (::Zygote.var"#41#42"{typeof(∂(#1))})(::Float64)@interface.jl:45
    gradient(::Function, ::Dict{Symbol,Float64})@interface.jl:54
    top-level scope@Local: 1
```

I am using Zygote 0.5.9

### PR DESCRIPTION
**Pull request by CarloLucibello**
Fix #817

### CODE REVIEW

File: src/lib/base.jl [hunk 1]
```diff
@@ -118,23 +118,26 @@ end
...
+    t0 = map(zero, t)
+    for (idx, v) in Δ
+      t0 = NamedTuple{N}(Base.setindex((t0...,), v, idx))
     end
-    return pairs(t), pairs_namedtuple
+    return (t0,)
+  end
+  
+  return pairs(t), pairs_namedtuple_pullback
 end
```
Comment (CarloLucibello) [inline src/lib/base.jl:133]: here just cleanup fixing indentation and renaming the pullback,

**Comment by CarloLucibello** [comment]
CI failures seem unrelated

File: src/lib/base.jl [hunk 2]
```diff
@@ -118,23 +118,26 @@ end
...
 
 @adjoint function Base.getfield(p::Pair, i::Int)
-    function pair_getfield(Δ)
-        f, s = i == 1 ? (Δ, zero(p[2])) : (zero(p[1]), Δ)
+    function pair_getfield_pullback(Δ)
+        f, s = i == 1 ? (Δ, nothing) : (nothing, Δ)
```
Comment (DhairyaLGandhi) [inline src/lib/base.jl:137]: could potentially cause keys to be overwritten in the dict, returning incorrect dictionaries. I had considered this for a previous #1060

Comment (CarloLucibello) [reply src/lib/base.jl:137]: I don't see how this is related to #1060 or can cause dictionary overwrite. This is fixing the adjoint of `getfield(p::Pair, i::Int)`

**Comment by CarloLucibello** [comment]
this is ready to go

**Pull request merged**